install_files/ansible-base/roles/common/tasks/remove_unused_packages.yml

```diff
@@ -8,6 +8,41 @@
     - apt
     - hardening
 
+# After installing securedrop-grsec, remove
+# old generic kernels to avoid accidental
+# boots into a less secure environment.
+# We must used command due to the use of wildcards
+- name: Remove generic kernel packages.
+  command: apt-get remove -y {{ item }}
+  with_items:
+    - linux-signed-generic
+    - linux-signed-generic-lts-utopic
+    - linux-signed-image-generic
+    - linux-signed-image-generic-lts-utopic
+    - linux-image-generic-lts-xenial
+    - 'linux-image-.*generic'
+    - 'linux-headers-.*'
+  register: apt_removed_kernels
+  changed_when: "'The following packages will be REMOVED' in apt_removed_kernels.stdout"
+  tags:
+    - apt
+
+- name: Get list of all installed kernels.
+  shell: dpkg-query -f '${Package} ${Status}\n' -W 'linux-image*' | awk '$NF == "installed"{print $1}'
+  register: apt_installed_kernels
+  changed_when: false
+  tags:
+    - apt
+
+- name: Validate that all installed kernels are grsecurity-hardened.
+  assert:
```

Comment:
@emkll i think this is fine for now but in the future this might have been easier/cleaner to just slap into a single grep command that exits 1 or 0